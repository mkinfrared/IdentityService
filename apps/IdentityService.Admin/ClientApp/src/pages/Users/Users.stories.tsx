import { Meta, StoryObj } from "@storybook/react";

import css from "./Story.module.scss";
import Users from "./Users";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/Users",
  component: Users,
  tags: ["autodocs"],
} satisfies Meta<typeof Users>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <Users {...args} />
    </div>
  ),
};

export { Default };

export default meta;
