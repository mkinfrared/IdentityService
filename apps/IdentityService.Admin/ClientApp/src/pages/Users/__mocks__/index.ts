export { default } from "./Users.mock";
