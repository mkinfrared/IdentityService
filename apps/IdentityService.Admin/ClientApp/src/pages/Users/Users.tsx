import css from "./Users.module.scss";
import { UsersProps } from "./Users.type";

const Users = ({ className }: UsersProps) => {
  const classNames = [css.Users, className].filter(Boolean).join(" ");

  return (
    <div className={classNames} data-testid="Users">
      Users works!!
    </div>
  );
};

export default Users;
