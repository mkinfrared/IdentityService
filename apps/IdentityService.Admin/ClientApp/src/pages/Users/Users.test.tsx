import { render } from "@testing-library/react";

import Users from "./Users";

describe("<Users />", () => {
  const Component = <Users />;

  it("should be defined", () => {
    expect(Users).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Users");

    expect(element).toBeDefined();
  });
});
