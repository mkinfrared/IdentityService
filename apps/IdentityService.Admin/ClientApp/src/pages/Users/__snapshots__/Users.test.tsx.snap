// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users /> should match the snapshot 1`] = `
<div>
  <div
    class="Users"
    data-testid="Users"
  >
    Users works!!
  </div>
</div>
`;
